rbx_types/src/qdir.rs

```diff
@@ -0,0 +1,106 @@
+use std::{
+    borrow::Borrow,
+    convert::Infallible,
+    ffi::{OsStr, OsString},
+    ops::Deref,
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
+#[derive(Debug, Clone, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(transparent)
+)]
+pub struct QDir {
+    path_buf: PathBuf,
+}
+
+impl QDir {
+    pub fn new() -> Self {
+        QDir {
+            path_buf: PathBuf::new(),
+        }
+    }
+
+    pub fn into_path_buf(self) -> PathBuf {
+        self.path_buf
+    }
+}
+
+impl AsRef<OsStr> for QDir {
+    fn as_ref(&self) -> &OsStr {
+        self.path_buf.as_os_str()
+    }
+}
+
+impl AsRef<Path> for QDir {
+    fn as_ref(&self) -> &Path {
+        self.path_buf.as_path()
+    }
+}
+
+impl Borrow<Path> for QDir {
+    fn borrow(&self) -> &Path {
+        self.deref()
+    }
+}
+
+impl Deref for QDir {
+    type Target = Path;
+
+    fn deref(&self) -> &Path {
+        self.path_buf.as_path()
+    }
+}
```

Comment:
I don't think a deref here is all that important. We can `AsRef` into `Path`, we're not making a total drop-in replacement.